code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -145,7 +145,10 @@
 	if(!status)
 		to_chat(user, "<span class='notice'>Secure the igniter first!</span>")
 		return
-	to_chat(user, "<span class='notice'>You ignite [src]!</span>")
+	if(!lit)
+		to_chat(user, "<span class='notice'>You ignite [src]!</span>")
```

Comment:
Colon is only a problem when you use to access an obj's var(like var/a = obj; a:name; instead of properly typecasting that var). Ternary operators are fine.